src/tests/unit/tests/background/permissions-state-tracker.test.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Interpreter } from 'background/interpreter';
+import { PermissionsStateTracker } from 'background/permissions-state-tracker';
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+import { Message } from 'common/message';
+import { Messages } from 'common/messages';
+import { tick } from 'tests/unit/common/tick';
+import { IMock, It, Mock, Times } from 'typemoq';
+
+describe('PermissionsStateTracker', () => {
+    let testSubject: PermissionsStateTracker;
+    let browserAdapterMock: IMock<BrowserAdapter>;
+    let interpreterMock: IMock<Interpreter>;
+
+    beforeEach(() => {
+        browserAdapterMock = Mock.ofType<BrowserAdapter>();
+        interpreterMock = Mock.ofType<Interpreter>();
+
+        testSubject = new PermissionsStateTracker(browserAdapterMock.object, interpreterMock.object);
+    });
+
+    it.each([true, false])('notifyChange sends the expected interpreter message', async permissionsState => {
```

Comment:
I think this test case is verifying implementation details; the user of the class doesn't really need to know about the private `notifyChange` function, they care that notifications are sent in response to browser events and `initialize()`.

To test the browser events, the usual pattern is to set up a mock browser adapter to record the onAddListener/onRemoveListener calls and save any listeners they received, then test the results of calling those saved listeners. You can either set up the mocks for this manually here, or you could extend the `SimulatedBrowserAdapter` we have for this purpose with logic similar to what it already does for many of the other `addListener` browserAdapter methods.